sqlx-core/src/any/connection/backend.rs

```diff
@@ -34,6 +34,14 @@ pub trait AnyConnectionBackend: std::any::Any + Debug + Send + 'static {
 
     fn start_rollback(&mut self);
 
+    /// Returns the current transaction depth.
+    ///
+    /// Transaction depth indicates the level of nested transactions:
+    /// - Level 0: No active transaction.
+    /// - Level 1: A transaction is active.
+    /// - Level 2 or higher: A transaction is active and one or more SAVEPOINTs have been created within it.
+    fn get_transaction_depth(&self) -> usize;
```

Comment:
Is this still BC?

Reply:
### Ignore `AnyConnectionBackend` Trait Compatibility

Making `get_transaction_depth()` a required method. If there is a custom implementation, the code will fail to compile unless the method is added.

```rust
pub trait AnyConnectionBackend: std::any::Any + Debug + Send + 'static {
    fn get_transaction_depth(&self) -> usize;  // This must be implemented
}
```

### Provide a Default Implementation in `AnyConnectionBackend` Trait Returning 0

This approach does not require `get_transaction_depth()` to be implemented, but it may return incorrect values in custom implementations that do not override the default method.

```rust
pub trait AnyConnectionBackend: std::any::Any + Debug + Send + 'static {
    fn get_transaction_depth(&self) -> usize {
        0 // Default implementation, may return incorrect values
    }
}
```

### Provide a Default Implementation in `AnyConnectionBackend` Trait Just Panics

Instead of returning `0`, it simply calls `panic!()`. However, this is highly undesirable because it turns what should be a static error into a runtime error.

```rust
pub trait AnyConnectionBackend: std::any::Any + Debug + Send + 'static {
    fn get_transaction_depth(&self) -> usize {
         panic!("Transaction depth is not implemented for this backend.")
    }
}
```

### Provide a Default Implementation in `AnyConnectionBackend` Using Downcasting with `Any`

This approach is not feasible because `sqlx-core` cannot depend on `sqlx-mysql`, `sqlx-postgres`, `sqlx-sqlite`, etc. Even if it were possible, there are gaps in the fallback logic, so it essentially just results in calling `panic!()` anyway. It unnecessarily complicates the implementation without adding real value.

```rust
pub trait AnyConnectionBackend: std::any::Any + Debug + Send + 'static {
    fn get_transaction_depth(&self) -> usize {
        let any: &dyn std::any::Any = self;
        if let Some(pg) = any.downcast_ref::<PgConnection>() {
            PgTransactionManager::get_transaction_depth(pg)
        } else if let Some(mysql) = any.downcast_ref::<MySqlConnection>() {
            MySqlTransactionManager::get_transaction_depth(mysql)
        } else if let Some(sqlite) = any.downcast_ref::<SqliteConnection>() {
            SqliteTransactionManager::get_transaction_depth(sqlite)
        } else {
            panic!("Transaction depth is not implemented for this backend.")
        }
    }
}
```

Reply:
In conclusion, it seems we have only two options: either target release version `0.9`, or allow this change in `0.8.*` specifically for `AnyConnectionBackend`, even though it might not be strictly SemVer-compliant.

Reply:
Fixed in 9409e5e